src/chain/elders_info.rs

```diff
@@ -120,22 +136,13 @@ impl EldersInfo {
 
     /// Returns `true` if the proofs are from a quorum of this section.
     pub fn is_quorum(&self, proofs: &ProofSet) -> bool {
-        // WIP: the call to members is probably to heavy?
-        proofs
-            .ids()
-            .filter(|id| self.members().contains(id))
-            .count()
-            * QUORUM_DENOMINATOR
-            > self.members.len() * QUORUM_NUMERATOR
+        proofs.ids().filter(|id| self.is_member(id)).count() * QUORUM_DENOMINATOR
+            > self.len() * QUORUM_NUMERATOR
     }
 
     /// Returns `true` if the proofs are from all members of this section.
     pub fn is_total_consensus(&self, proofs: &ProofSet) -> bool {
-        proofs
-            .ids()
-            .filter(|id| self.members().contains(id))
-            .count()
-            == self.members.len()
+        proofs.ids().filter(|id| self.is_member(id)).count() == self.members.len()
```

Comment:
not consistent with is_quorum (ideally would use `self.len()` as well. (Also keep level of abstraction consistent which is good).